### PR TITLE
feat: Adding support for benchmark init command

### LINKED ISSUE #129: Add new command for generating executableTarget and source boilerplate for new benchmarks
Something like

`swift package benchmark init MyBenchmark`

1. creates the `Benchmarks/MyBenchmark` directory

2. creates the `Benchmarks/MyBenchmark/MyBenchmark.swift` source file with the boilerplate:

```
import Benchmark
import Foundation

let benchmarks = {
    Benchmark.defaultConfiguration = .init(scalingFactor: .kilo)

    Benchmark("SomeBenchmark") { benchmark in
        for _ in benchmark.scaledIterations {
            blackHole(Date()) // replace this line with your own benchmark
        }
    }
}
```
3. generates this to standard out (preferably so we can pipe it to `pbcopy`) :

```
        // MyBenchmark benchmark target
        .executableTarget(
            name: "MyBenchmark",
            dependencies: [
                .product(name: "Benchmark", package: "package-benchmark"),
                .product(name: "BenchmarkPlugin", package: "package-benchmark"),
            ],
            path: "Benchmarks/MyBenchmark"
        ),
```

Which can then just be copy-pasted into Package.swift and we're ready to go

### PR DESCRIPTION
**Pull request by hassila**
## Description

Fixes #129 

Adds `swift package benchmark init <target>` supporting creation of boilerplate + adding target to Package.swift.

## How Has This Been Tested?

Manual testing.

## Minimal checklist:

- [x] I have performed a self-review of my own code 
- [x] I have added `DocC` code-level documentation for any public interfaces exported by the package
- [ ] I have added unit and/or integration tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: Plugins/BenchmarkTool/BenchmarkTool+CreateBenchmark.swift [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        subPath.append("\(targetName)") // package/Benchmarks/targetName
+
+        outputPath.createSubPath(subPath) // Create destination subpath if needed
+        outputPath.append(subPath.components)
+        outputPath.append("\(targetName).swift")
+
+        // Write out benchmark boilerplate, open with .exclusiveCreate to abort if file exists
+        do {
+            let fd = try FileDescriptor.open(
+                outputPath, .writeOnly, options: [.truncate, .create, .exclusiveCreate], permissions: .ownerReadWrite
```
Comment (dimlio) [inline Plugins/BenchmarkTool/BenchmarkTool+CreateBenchmark.swift:121]: Need `.truncate` if we expect to create a new file?

**Review by dimlio — APPROVED** [review]

File: Plugins/BenchmarkTool/BenchmarkTool+CreateBenchmark.swift (hunk 1 above)
Comment (hassila) [reply Plugins/BenchmarkTool/BenchmarkTool+CreateBenchmark.swift:121]: `.truncate` is there?

Comment (hassila) [reply Plugins/BenchmarkTool/BenchmarkTool+CreateBenchmark.swift:121]: ` [.truncate, .create, .exclusiveCreate]`

Comment (dimlio) [reply Plugins/BenchmarkTool/BenchmarkTool+CreateBenchmark.swift:121]: yep, but is it needed? [](edited-with-Axolo)

Comment (hassila) [reply Plugins/BenchmarkTool/BenchmarkTool+CreateBenchmark.swift]: Ah, no. Removed it.

**Pull request merged**